### PR TITLE
Refactor eventbridge client

### PR DESCRIPTION
**Pull request by MancunianSam**
This will allow us to make test implementations of the trait.

### CODE REVIEW

File: eventbridge/src/main/scala/uk/gov/nationalarchives/DAEventBridgeClient.scala [hunk 1]
```diff
@@ -59,7 +44,24 @@ object DAEventBridgeClient:
...
+          .entries(requestEntry)
+          .build()
+
+        Async[F].fromCompletableFuture(Async[F].pure(eventBridgeAsyncClient.putEvents(putEventsRequest)))
+      }
+
+    }
 
-  def apply[F[_]: Async]() = new DAEventBridgeClient[F](asyncClient)
+  def apply[F[_]: Async](): DAEventBridgeClient[F] = DAEventBridgeClient[F](asyncClient)
```
Comment (techncl) [inline eventbridge/src/main/scala/uk/gov/nationalarchives/DAEventBridgeClient.scala:67]: Do we need this? could we not just add a default argument to line 42, e.g. `def apply[F[_]: Async](eventBridgeAsyncClient: EventBridgeAsyncClient=asyncClient): DAEventBridgeClient[F]`?

**Review by techncl — COMMENTED** [review]
Looks good! Just one comment. Also, could you update the `index.md`?

File: eventbridge/src/main/scala/uk/gov/nationalarchives/DAEventBridgeClient.scala (hunk 1 above)
Comment (MancunianSam) [reply eventbridge/src/main/scala/uk/gov/nationalarchives/DAEventBridgeClient.scala]: Yeah, ok. I've changed it.

**Review by techncl — APPROVED** [review]
👍

**Pull request merged**